gapic/src/main/com/google/gapid/perfetto/models/FrameEventsTrack.java

```diff
@@ -333,6 +391,59 @@ public void getRange(Consumer<TimeSpan> span) {
         slice.getRange(span);
       }
     }
+
+    public FrameSelection getSelection() {
+      return frameSelection;
+    }
+  }
+
+  public static class FrameSelection {
+    public static FrameSelection EMPTY = new FrameSelection();
+    // key = concat(layerName, '_', frameNumber)
+    private List<String> keys;
+
+    public FrameSelection() {
+      keys = Lists.newArrayList();
+    }
+
+    public FrameSelection(long[] f, String[] l) {
+      keys = Lists.newArrayList();
+      for (int i = 0; i < l.length; i++) {
+        keys.add(l[i] + "_" + f[i]);
+      }
+    }
+
+    public boolean contains(long[] f, String[] l) {
+      for (int i = 0; i < f.length; i++) {
+        if (keys.contains(l[i] + "_" + f[i])) {
```

Comment:
It looks like we can change `List<String> keys` to `Set<String> keys` to improve efficiency this line. Or maybe I'm missing sth and there's a reason using list here?

Reply:
Done.